packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -104,9 +103,29 @@ class DatePickerInput extends React.Component {
   constructor(props) {
     super(props);
 
+    const format = DateUtil.getDateFormat(props.intl.formatMessage({ id: 'Terra.datePicker.dateFormatOrder' }));
+    const tempDate = DateUtil.convertToISO8601(props.value, format);
+
+    this.state = {};
+
+    if (tempDate) {
+      this.state = {
+        day: tempDate.substr(8, 2),
```

Comment:
I could be wrong but if the return value of [`DateUtil.convertToISO8601()`](https://github.com/cerner/terra-framework/blob/809ece57b4400499b0200f7bd9a34445f886dded/packages/terra-date-picker/src/DateUtil.js#L121-L127) is "invalid", then aren't we parsing and sticking to incorrect values here?

Reply:
This has been mostly removed and adjusted in various ways: [89c3693](https://github.com/cerner/terra-framework/pull/1004/commits/89c3693f31162adfc53f90f681cdfda21588b6b0)